security/advancedtls/advancedtls_test.go

```diff
@@ -59,17 +133,15 @@ type fakeProvider struct {
 	isClient      bool
 	wantMultiCert bool
 	wantError     bool
+	t             *testing.T
```

Comment:
This doesn't look right to me. You should not have to plumb the `testing.T` all the way down here. Maybe change `loadCerts` to not take `testing.T` as a parameter and instead make it return an error.